File: ui/script/trigger_validator.ts
module validator {
	//$$.gameplay.dflt.default_units_ts_ChargeSkillShowMethod:Data.Game.ChargeCoolDownMethod
	/** @noSelf */
	export let validator_119251571:(this: void, 对象:object)=>number = (this: void, 对象:object):number => {
		let 技能: Skill = base.force_as<Skill>(对象);
		let result: number = 技能.get_currrent_charge_show_cd();
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ChargeSkillShowMethod:Data.Game.ChargeMaxCoolDownMethod
	/** @noSelf */
	export let validator_24819615:(this: void, 对象:object)=>number = (this: void, 对象:object):number => {
		let 技能: Skill = base.force_as<Skill>(对象);
		let result: number = 技能.get_max_charge_show_cd();
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod:Data.Game.ShowNameMethod
	/** @noSelf */
	export let validator_9155707:(this: void, 对象:object)=>string = (this: void, 对象:object):string => {
		let 单位: Unit = base.force_as<Unit>(对象);
		let result: string = 单位.get_display_name();
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod:Data.Game.TipsMethod
	/** @noSelf */
	export let validator_135285719:(this: void, 对象:object)=>string = (this: void, 对象:object):string => {
		let 单位: Unit = base.force_as<Unit>(对象);
		let result: string = "无描述";
		if ((((单位.cache.Description != undefined) && (单位.cache.Description != "")))) {
			result = base.i18n.get_text_ex(单位.cache.Description, base.i18n.get_lang());
		} else {
		}
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod:Data.Game.IconMethod
	/** @noSelf */
	export let validator_25166946:(this: void, 对象:object)=>string = (this: void, 对象:object):string => {
		let 单位: Unit = base.force_as<Unit>(对象);
		let result: string = 单位.cache.Icon;
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod:Data.Game.MaxCoolDownMethod
	/** @noSelf */
	export let validator_25626537:(this: void, 对象:object)=>number = (this: void, 对象:object):number => {
		let result: number = 0;
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod:Data.Game.CoolDownMethod
	/** @noSelf */
	export let validator_216264469:(this: void, 对象:object)=>number = (this: void, 对象:object):number => {
		let result: number = 0;
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod_1:Data.Game.ShowNameMethod
	/** @noSelf */
	export let validator_240778512:(this: void, 对象:object)=>string = (this: void, 对象:object):string => {
		let 物品: Item = base.force_as<Item>(对象);
		let result: string = base.i18n.get_text_ex(物品.cache.Name, base.i18n.get_lang());
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod_1:Data.Game.TipsMethod
	/** @noSelf */
	export let validator_268327356:(this: void, 对象:object)=>string = (this: void, 对象:object):string => {
		let 物品: Item = base.force_as<Item>(对象);
		let result: string = "无描述";
		if (((物品.cache.Description != undefined))&&((物品.cache.Description != ""))) {
			result = base.i18n.get_text_ex(物品.cache.Description, base.i18n.get_lang());
		} else {
		}
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod_1:Data.Game.IconMethod
	/** @noSelf */
	export let validator_149965755:(this: void, 对象:object)=>string = (this: void, 对象:object):string => {
		let 物品: Item = base.force_as<Item>(对象);
		let result: string = 物品.cache.Icon;
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod_1:Data.Game.MaxCoolDownMethod
	/** @noSelf */
	export let validator_51091782:(this: void, 对象:object)=>number = (this: void, 对象:object):number => {
		let result: number = 0;
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod_1:Data.Game.CoolDownMethod
	/** @noSelf */
	export let validator_228843244:(this: void, 对象:object)=>number = (this: void, 对象:object):number => {
		let result: number = 0;
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod_2:Data.Game.ShowNameMethod
	/** @noSelf */
	export let validator_40883262:(this: void, 对象:object)=>string = (this: void, 对象:object):string => {
		let 技能: Skill = base.force_as<Skill>(对象);
		let result: string = base.i18n.get_text_ex(技能.cache.Name, base.i18n.get_lang());
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod_2:Data.Game.TipsMethod
	/** @noSelf */
	export let validator_242268571:(this: void, 对象:object)=>string = (this: void, 对象:object):string => {
		let 技能: Skill = base.force_as<Skill>(对象);
		let result: string = 技能.get_tip();
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod_2:Data.Game.IconMethod
	/** @noSelf */
	export let validator_219755542:(this: void, 对象:object)=>string = (this: void, 对象:object):string => {
		let 技能: Skill = base.force_as<Skill>(对象);
		let result: string = 技能.cache.IconName;
		let 开关技能数编表: _OBJ__spell_SpellToggle | undefined = base.force_as<_OBJ__spell_SpellToggle>(技能.cache);
		let 段数: number = 0;
		let 多段图标: string = "字符串";
		if ((技能.is_toggled_on())) {
			if (((开关技能数编表.IconNameOff != undefined))&&((开关技能数编表.IconNameOff != ""))&&((开关技能数编表.IconNameOff != "unkown"))) {
				result = 开关技能数编表.IconNameOff;
			} else {
			}
		} else {
		}
		if (((技能.cache.MultiPhaseSetting != undefined))&&((技能.cache.MultiPhaseSetting.IsMultiPhase == true))&&((技能.cache.MultiPhaseSetting.MultiPhaseConfig != undefined))) {
			段数 = base.math.min(技能.cache.MultiPhaseSetting.MultiPhaseConfig.length, 技能.get_phase());
			段数 = (段数 - 1);
			多段图标 = 技能.cache.MultiPhaseSetting.MultiPhaseConfig[段数].Icon;
			if (((多段图标 != undefined))&&((多段图标 != ""))&&((多段图标 != "unkown"))) {
				result = 多段图标;
			} else {
			}
		} else {
		}
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod_2:Data.Game.MaxCoolDownMethod
	/** @noSelf */
	export let validator_14998893:(this: void, 对象:object)=>number = (this: void, 对象:object):number => {
		let 技能: Skill = base.force_as<Skill>(对象);
		let result: number = 技能.get_max_show_cd();
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod_2:Data.Game.CoolDownMethod
	/** @noSelf */
	export let validator_117750349:(this: void, 对象:object)=>number = (this: void, 对象:object):number => {
		let 技能: Skill = base.force_as<Skill>(对象);
		let result: number = 技能.get_current_show_cd();
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod_3:Data.Game.ShowNameMethod
	/** @noSelf */
	export let validator_75992055:(this: void, 对象:object)=>string = (this: void, 对象:object):string => {
		let Buff: Buff = base.force_as<Buff>(对象);
		let result: string = base.i18n.get_text_ex(Buff.cache.Name, base.i18n.get_lang());
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod_3:Data.Game.TipsMethod
	/** @noSelf */
	export let validator_177744544:(this: void, 对象:object)=>string = (this: void, 对象:object):string => {
		let Buff: Buff = base.force_as<Buff>(对象);
		let result: string = "无描述";
		if (((Buff.cache.Description != undefined))&&((Buff.cache.Description != ""))) {
			result = base.i18n.get_text_ex(Buff.cache.Description, base.i18n.get_lang());
		} else {
		}
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod_3:Data.Game.IconMethod
	/** @noSelf */
	export let validator_131864409:(this: void, 对象:object)=>string = (this: void, 对象:object):string => {
		let Buff: Buff = base.force_as<Buff>(对象);
		let result: string = Buff.cache.BuffIcon;
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod_3:Data.Game.MaxCoolDownMethod
	/** @noSelf */
	export let validator_235454563:(this: void, 对象:object)=>number = (this: void, 对象:object):number => {
		let Buff: Buff = base.force_as<Buff>(对象);
		let result: number = Buff.get_time();
		return result;
	}

	//$$.gameplay.dflt.default_units_ts_ObjectShowMethod_3:Data.Game.CoolDownMethod
	/** @noSelf */
	export let validator_102943768:(this: void, 对象:object)=>number = (this: void, 对象:object):number => {
		let Buff: Buff = base.force_as<Buff>(对象);
		let result: number = Buff.get_remaining();
		return result;
	}

	//$$default_units_ts.unit.火龙.MoverFunction:Data.Game.FunctionClient_UI
	/** @noSelf */
	export let validator_38385052:(this: void, 移动器Id:IdPreset<"mover_id">, 流逝时间:number, 弹道单位:Unit, 施法者单位:Unit, 目标单位:Unit, 发射点:Point, 目标点:Point)=>void = (this: void, 移动器Id:IdPreset<"mover_id">, 流逝时间:number, 弹道单位:Unit, 施法者单位:Unit, 目标单位:Unit, 发射点:Point, 目标点:Point):void => {
		let 半径: number = (base.point_distance(发射点, 目标点) * 0.5);
		let 初始角度: number = base.point_angle(目标点, 发射点);
		let 圆心: Point = base.point_move(目标点, 初始角度, 半径);
		let 当前位置: Point = base.point_move(圆心, (初始角度 + (流逝时间 * 100)), 半径);
		base.set_unit_location_and_height(弹道单位, 当前位置, 150);
	}

	/////商店按钮点击事件.lua
	/** @noSelf */
	export let validator_199041480:(this: void, ctrl:_OBJ__gui_ctrl_button, mouse_btn_id:number)=>void = (this: void, ctrl:_OBJ__gui_ctrl_button, mouse_btn_id:number):void => {
		base.send_custom_event(new p_8e93.点击商店按钮(((base.game) as unknown as base.Game)));
	}

	/////开始按钮点击事件.lua
	/** @noSelf */
	export let validator_36669552:(this: void, ctrl:_OBJ_gameui_gui_ctrl_normal_btn, mouse_btn_id:number)=>void = (this: void, ctrl:_OBJ_gameui_gui_ctrl_normal_btn, mouse_btn_id:number):void => {
		base.send_custom_event(new p_8e93.点击开始游戏按钮(((base.game) as unknown as base.Game)));
		ctrl.show = false;
	}

	/////防御塔升级按钮点击事件.lua
	/** @noSelf */
	export let validator_243098681:(this: void, ctrl:_OBJ_gameui_gui_ctrl_normal_btn, mouse_btn_id:number)=>void = (this: void, ctrl:_OBJ_gameui_gui_ctrl_normal_btn, mouse_btn_id:number):void => {
		base.send_custom_event(new p_8e93.点击防御塔升级按钮(((base.game) as unknown as base.Game)));
	}

	/////防御塔拆除按钮点击事件.lua
	/** @noSelf */
	export let validator_137169674:(this: void, ctrl:_OBJ_gameui_gui_ctrl_normal_btn, mouse_btn_id:number)=>void = (this: void, ctrl:_OBJ_gameui_gui_ctrl_normal_btn, mouse_btn_id:number):void => {
		base.send_custom_event(new p_8e93.点击防御塔拆除按钮(((base.game) as unknown as base.Game)));
	}

	//init validators
	function init_validator_0() {
		let data: any = undefined
		if (data = base.eff.cache("$$.gameplay.dflt.default_units_ts_ChargeSkillShowMethod")) {
			data.ChargeCoolDownMethod = validator_119251571
			data.ChargeMaxCoolDownMethod = validator_24819615
		}
		if (data = base.eff.cache("$$.gameplay.dflt.default_units_ts_ObjectShowMethod")) {
			data.ShowNameMethod = validator_9155707
			data.TipsMethod = validator_135285719
			data.IconMethod = validator_25166946
			data.MaxCoolDownMethod = validator_25626537
			data.CoolDownMethod = validator_216264469
		}
		if (data = base.eff.cache("$$.gameplay.dflt.default_units_ts_ObjectShowMethod_1")) {
			data.ShowNameMethod = validator_240778512
			data.TipsMethod = validator_268327356
			data.IconMethod = validator_149965755
			data.MaxCoolDownMethod = validator_51091782
			data.CoolDownMethod = validator_228843244
		}
		if (data = base.eff.cache("$$.gameplay.dflt.default_units_ts_ObjectShowMethod_2")) {
			data.ShowNameMethod = validator_40883262
			data.TipsMethod = validator_242268571
			data.IconMethod = validator_219755542
			data.MaxCoolDownMethod = validator_14998893
			data.CoolDownMethod = validator_117750349
		}
		if (data = base.eff.cache("$$.gameplay.dflt.default_units_ts_ObjectShowMethod_3")) {
			data.ShowNameMethod = validator_75992055
			data.TipsMethod = validator_177744544
			data.IconMethod = validator_131864409
			data.MaxCoolDownMethod = validator_235454563
			data.CoolDownMethod = validator_102943768
		}
		if (data = base.eff.cache("$$default_units_ts.unit.火龙.MoverFunction")) {
			data.FunctionClient_UI = validator_38385052
		}
	}
	init_validator_0()
}

File: ui/script/trigger_validator.lua
local ____lualib = base.tsc
local __TS__TypeReference = ____lualib.__TS__TypeReference
local __TS__New = ____lualib.__TS__New
local __TS__Class2 = ____lualib.__TS__Class2
RegionCircle = base.tsc.CLASSES.RegionCircle or __TS__Class2("RegionCircle")
RegionRect = base.tsc.CLASSES.RegionRect or __TS__Class2("RegionRect")
Point = base.tsc.CLASSES.Point or __TS__Class2("Point")
Unit = base.tsc.CLASSES.Unit or __TS__Class2("Unit")
Riseletter = base.tsc.CLASSES.Riseletter or __TS__Class2("Riseletter")
Region = base.tsc.CLASSES.Region or __TS__Class2("Region")
Vector = base.tsc.CLASSES.Vector or __TS__Class2("Vector")
Slot = base.tsc.CLASSES.Slot or __TS__Class2("Slot")
Camera = base.tsc.CLASSES.Camera or __TS__Class2("Camera")
DataCache = base.tsc.CLASSES.DataCache or __TS__Class2("DataCache")
YPR旋转 = base.tsc.CLASSES.YPR旋转 or __TS__Class2("YPR旋转")
EffectParamShared = base.tsc.CLASSES.EffectParamShared or __TS__Class2("EffectParamShared")
EffectParam = base.tsc.CLASSES.EffectParam or __TS__Class2("EffectParam")
Target = base.tsc.CLASSES.Target or __TS__Class2("Target")
ScreenPos = base.tsc.CLASSES.ScreenPos or __TS__Class2("ScreenPos")
Actor = base.tsc.CLASSES.Actor or __TS__Class2("Actor")
UnitGroup = base.tsc.CLASSES.UnitGroup or __TS__Class2("UnitGroup")
TriggerEvent = base.tsc.CLASSES.TriggerEvent or __TS__Class2("TriggerEvent")
TriggerParamTable = base.tsc.CLASSES.TriggerParamTable or __TS__Class2("TriggerParamTable")
Player = base.tsc.CLASSES.Player or __TS__Class2("Player")
Skill = base.tsc.CLASSES.Skill or __TS__Class2("Skill")
Buff = base.tsc.CLASSES.Buff or __TS__Class2("Buff")
Timer = base.tsc.CLASSES.Timer or __TS__Class2("Timer")
Item = base.tsc.CLASSES.Item or __TS__Class2("Item")
Trigger = base.tsc.CLASSES.Trigger or __TS__Class2("Trigger")
Coroutine = base.tsc.CLASSES.Coroutine or __TS__Class2("Coroutine")
IEventNotify = base.tsc.CLASSES.IEventNotify or __TS__Class2("IEventNotify")
Present = base.tsc.CLASSES.Present or __TS__Class2("Present")
Array = base.tsc.CLASSES.Array or __TS__Class2("Array")
validator = validator or ({})
do
    ---
    -- @noSelf
    validator.validator_119251571 = function(对象)
        local 技能 = base.force_as(
            __TS__TypeReference(Skill, {}),
            对象
        )
        local result = 技能:get_currrent_charge_show_cd()
        return result
    end
    ---
    -- @noSelf
    validator.validator_24819615 = function(对象)
        local 技能 = base.force_as(
            __TS__TypeReference(Skill, {}),
            对象
        )
        local result = 技能:get_max_charge_show_cd()
        return result
    end
    ---
    -- @noSelf
    validator.validator_9155707 = function(对象)
        local 单位 = base.force_as(
            __TS__TypeReference(Unit, {}),
            对象
        )
        local result = 单位:get_display_name()
        return result
    end
    ---
    -- @noSelf
    validator.validator_135285719 = function(对象)
        local 单位 = base.force_as(
            __TS__TypeReference(Unit, {}),
            对象
        )
        local result = "无描述"
        if 单位.cache.Description ~= nil and 单位.cache.Description ~= "" then
            result = base.i18n.get_text_ex(
                单位.cache.Description,
                base.i18n.get_lang()
            )
        else
        end
        return result
    end
    ---
    -- @noSelf
    validator.validator_25166946 = function(对象)
        local 单位 = base.force_as(
            __TS__TypeReference(Unit, {}),
            对象
        )
        local result = 单位.cache.Icon
        return result
    end
    ---
    -- @noSelf
    validator.validator_25626537 = function(对象)
        local result = 0
        return result
    end
    ---
    -- @noSelf
    validator.validator_216264469 = function(对象)
        local result = 0
        return result
    end
    ---
    -- @noSelf
    validator.validator_240778512 = function(对象)
        local 物品 = base.force_as(
            __TS__TypeReference(Item, {}),
            对象
        )
        local result = base.i18n.get_text_ex(
            物品.cache.Name,
            base.i18n.get_lang()
        )
        return result
    end
    ---
    -- @noSelf
    validator.validator_268327356 = function(对象)
        local 物品 = base.force_as(
            __TS__TypeReference(Item, {}),
            对象
        )
        local result = "无描述"
        if 物品.cache.Description ~= nil and 物品.cache.Description ~= "" then
            result = base.i18n.get_text_ex(
                物品.cache.Description,
                base.i18n.get_lang()
            )
        else
        end
        return result
    end
    ---
    -- @noSelf
    validator.validator_149965755 = function(对象)
        local 物品 = base.force_as(
            __TS__TypeReference(Item, {}),
            对象
        )
        local result = 物品.cache.Icon
        return result
    end
    ---
    -- @noSelf
    validator.validator_51091782 = function(对象)
        local result = 0
        return result
    end
    ---
    -- @noSelf
    validator.validator_228843244 = function(对象)
        local result = 0
        return result
    end
    ---
    -- @noSelf
    validator.validator_40883262 = function(对象)
        local 技能 = base.force_as(
            __TS__TypeReference(Skill, {}),
            对象
        )
        local result = base.i18n.get_text_ex(
            技能.cache.Name,
            base.i18n.get_lang()
        )
        return result
    end
    ---
    -- @noSelf
    validator.validator_242268571 = function(对象)
        local 技能 = base.force_as(
            __TS__TypeReference(Skill, {}),
            对象
        )
        local result = 技能:get_tip()
        return result
    end
    ---
    -- @noSelf
    validator.validator_219755542 = function(对象)
        local 技能 = base.force_as(
            __TS__TypeReference(Skill, {}),
            对象
        )
        local result = 技能.cache.IconName
        local 开关技能数编表 = base.force_as(
            __TS__TypeReference(_OBJ__spell_SpellToggle, {}),
            技能.cache
        )
        local 段数 = 0
        local 多段图标 = "字符串"
        if 技能:is_toggled_on() then
            if 开关技能数编表.IconNameOff ~= nil and 开关技能数编表.IconNameOff ~= "" and 开关技能数编表.IconNameOff ~= "unkown" then
                result = 开关技能数编表.IconNameOff
            else
            end
        else
        end
        if 技能.cache.MultiPhaseSetting ~= nil and 技能.cache.MultiPhaseSetting.IsMultiPhase == true and 技能.cache.MultiPhaseSetting.MultiPhaseConfig ~= nil then
            段数 = base.math.min(
                #技能.cache.MultiPhaseSetting.MultiPhaseConfig,
                技能:get_phase()
            )
            段数 = 段数 - 1
            多段图标 = 技能.cache.MultiPhaseSetting.MultiPhaseConfig[段数 + 1].Icon
            if 多段图标 ~= nil and 多段图标 ~= "" and 多段图标 ~= "unkown" then
                result = 多段图标
            else
            end
        else
        end
        return result
    end
    ---
    -- @noSelf
    validator.validator_14998893 = function(对象)
        local 技能 = base.force_as(
            __TS__TypeReference(Skill, {}),
            对象
        )
        local result = 技能:get_max_show_cd()
        return result
    end
    ---
    -- @noSelf
    validator.validator_117750349 = function(对象)
        local 技能 = base.force_as(
            __TS__TypeReference(Skill, {}),
            对象
        )
        local result = 技能:get_current_show_cd()
        return result
    end
    ---
    -- @noSelf
    validator.validator_75992055 = function(对象)
        local Buff = base.force_as(
            __TS__TypeReference(Buff, {}),
            对象
        )
        local result = base.i18n.get_text_ex(
            Buff.cache.Name,
            base.i18n.get_lang()
        )
        return result
    end
    ---
    -- @noSelf
    validator.validator_177744544 = function(对象)
        local Buff = base.force_as(
            __TS__TypeReference(Buff, {}),
            对象
        )
        local result = "无描述"
        if Buff.cache.Description ~= nil and Buff.cache.Description ~= "" then
            result = base.i18n.get_text_ex(
                Buff.cache.Description,
                base.i18n.get_lang()
            )
        else
        end
        return result
    end
    ---
    -- @noSelf
    validator.validator_131864409 = function(对象)
        local Buff = base.force_as(
            __TS__TypeReference(Buff, {}),
            对象
        )
        local result = Buff.cache.BuffIcon
        return result
    end
    ---
    -- @noSelf
    validator.validator_235454563 = function(对象)
        local Buff = base.force_as(
            __TS__TypeReference(Buff, {}),
            对象
        )
        local result = Buff:get_time()
        return result
    end
    ---
    -- @noSelf
    validator.validator_102943768 = function(对象)
        local Buff = base.force_as(
            __TS__TypeReference(Buff, {}),
            对象
        )
        local result = Buff:get_remaining()
        return result
    end
    ---
    -- @noSelf
    validator.validator_38385052 = function(移动器Id, 流逝时间, 弹道单位, 施法者单位, 目标单位, 发射点, 目标点)
        local 半径 = base.point_distance(发射点, 目标点) * 0.5
        local 初始角度 = base.point_angle(目标点, 发射点)
        local 圆心 = base.point_move(目标点, 初始角度, 半径)
        local 当前位置 = base.point_move(圆心, 初始角度 + 流逝时间 * 100, 半径)
        base.set_unit_location_and_height(弹道单位, 当前位置, 150)
    end
    ---
    -- @noSelf
    validator.validator_199041480 = function(ctrl, mouse_btn_id)
        base.send_custom_event(__TS__New(p_8e93.点击商店按钮, {}, base.game))
    end
    ---
    -- @noSelf
    validator.validator_36669552 = function(ctrl, mouse_btn_id)
        base.send_custom_event(__TS__New(p_8e93.点击开始游戏按钮, {}, base.game))
        ctrl.show = false
    end
    ---
    -- @noSelf
    validator.validator_243098681 = function(ctrl, mouse_btn_id)
        base.send_custom_event(__TS__New(p_8e93.点击防御塔升级按钮, {}, base.game))
    end
    ---
    -- @noSelf
    validator.validator_137169674 = function(ctrl, mouse_btn_id)
        base.send_custom_event(__TS__New(p_8e93.点击防御塔拆除按钮, {}, base.game))
    end
    local function init_validator_0(self)
        local data = nil
        data = base.eff.cache("$$.gameplay.dflt.default_units_ts_ChargeSkillShowMethod")
        if data then
            data.ChargeCoolDownMethod = validator.validator_119251571
            data.ChargeMaxCoolDownMethod = validator.validator_24819615
        end
        data = base.eff.cache("$$.gameplay.dflt.default_units_ts_ObjectShowMethod")
        if data then
            data.ShowNameMethod = validator.validator_9155707
            data.TipsMethod = validator.validator_135285719
            data.IconMethod = validator.validator_25166946
            data.MaxCoolDownMethod = validator.validator_25626537
            data.CoolDownMethod = validator.validator_216264469
        end
        data = base.eff.cache("$$.gameplay.dflt.default_units_ts_ObjectShowMethod_1")
        if data then
            data.ShowNameMethod = validator.validator_240778512
            data.TipsMethod = validator.validator_268327356
            data.IconMethod = validator.validator_149965755
            data.MaxCoolDownMethod = validator.validator_51091782
            data.CoolDownMethod = validator.validator_228843244
        end
        data = base.eff.cache("$$.gameplay.dflt.default_units_ts_ObjectShowMethod_2")
        if data then
            data.ShowNameMethod = validator.validator_40883262
            data.TipsMethod = validator.validator_242268571
            data.IconMethod = validator.validator_219755542
            data.MaxCoolDownMethod = validator.validator_14998893
            data.CoolDownMethod = validator.validator_117750349
        end
        data = base.eff.cache("$$.gameplay.dflt.default_units_ts_ObjectShowMethod_3")
        if data then
            data.ShowNameMethod = validator.validator_75992055
            data.TipsMethod = validator.validator_177744544
            data.IconMethod = validator.validator_131864409
            data.MaxCoolDownMethod = validator.validator_235454563
            data.CoolDownMethod = validator.validator_102943768
        end
        data = base.eff.cache("$$default_units_ts.unit.火龙.MoverFunction")
        if data then
            data.FunctionClient_UI = validator.validator_38385052
        end
    end
    init_validator_0(nil)
end
